### ISSUE TITLE
Dataset

### ISSUE THREAD

**Issue by villmow** [guxd/deep-code-search#2]
Hi guys, awesome project. Would you mind releasing the original training and testing dataset, without any pickled or preprocessed files?

**Comment by matanpugach** [comment]
@0bserver07 @guxd could you share the raw method names\descriptions?

**Comment by villmow** [comment]
@matanpugach That's what I meant! 

I already downloaded the files from Gdrive, but they are preprocessed. Everyone using your dataset is limited to your features (token, api sequence, name tokens) and a vocabulary limit of 10,000. One cannot restore the original "RawCode" - "Documentation" mapping from your dataset, to - for example - try new features.

**Comment by wanyao1992** [comment]
Hi @villmow,
I have the same requirement, have you got the raw datasets without preprocessing from @guxd?

**Comment by gauravkoradiya** [comment]
i have python raw data how to preprocess it as same u did for java. I would to build code search for python code. How to do preprocessing??

**Comment by guxd** [comment]
@gauravkoradiya You should use python code parser. Python provides an `ast` module which supports the parsing.  
Here is a sample project which parses python code into ASTs. 
https://github.com/fyrestone/pycode_similar
You may find more from the GitHub. 
After that, you need to convert ASTs into call sequences.

**Comment by gauravkoradiya** [comment]
> @gauravkoradiya You should use python code parser. Python provides an `ast` module which supports the parsing.
> Here is a sample project which parses python code into ASTs.
> https://github.com/fyrestone/pycode_similar
> You may find more from the GitHub.
> After that, you need to convert ASTs into call sequences.

**Comment by gauravkoradiya** [comment]
Awesome..thank you....I got it.

**Comment by hoogang** [comment]
could you share the original datasets  without any pickled or preprocessed files?

**Comment by jackalhan** [comment]
I agree with @hoogang . @guxd , would you please release your raw or original datasets without any pickled or preprocessed files?
Thank you.

**Comment by guxd** [comment]
The raw code datasets are available at `/pytorch/train.rawcode.rar`
